### DISCUSSION TITLE
How to use `TaskDialog` with `ReactiveUI` ?

### DISCUSSION THREAD

**Discussion by Crauzer** [amwx/FluentAvalonia#269] [Q&A]
Hello! I wanted to ask if anybody knows what is the correct way to use the `TaskDialog` together with `ReactiveUI`, specifically, showing it from a `ReactiveCommand`/`Interaction`.
I tried running the dialog from an `Interaction` in the view but after the interaction finishes, an exception is thrown "Failed to find a registration for a Interaction.".
Any help would be appreciated!
If I find a solution to this, I'll post it here for others.

**Comment by amwx** [comment]
So I don't know much about RxUI so I probably can't be of too much help (sorry!), but from a quick look at the [documentation on interactions](https://www.reactiveui.net/docs/handbook/interactions/) it doesn't look like there needs to be anything special to make it work (so that's good). In other RxUI questions, the usual cause for errors has always been using the wrong thread. Make sure you're using the Avalonia RxUI scheduler and not the default one.

**Comment by mgnslndh** [comment]
`TaskDialog` works great with `Interaction` from `ReactiveUI`. Make sure you follow the guidelines from the [documentation](https://www.reactiveui.net/docs/handbook/interactions/). The view needs to register a handler for the interaction and the handler is then responsible for showing the `TaskDialog`
